compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -816,7 +817,8 @@ def create_or_choose_auth_project(ac: AuthClient) -> str:
     Lets the user choose from one of their existing projects, or
     prompt for a description and creates one for them
     """
-    projects = ac.get_projects()
+    assert ac._app is not None  # mypy
```

Comment:
You can avoid these asserts with `app = get_globus_app_with_scopes()`. You might also consider moving this directly in the `gare_handler()` function:
```py
def gare_handler(f: t.Callable, *args, **kwargs):
    app = get_globus_app_with_scopes()
    ...
```

Reply:
from my testing, if you don't pass the `AuthClient`'s app and login with a different app instance before re-trying the client method, the client's stale app instance doesn't pick up the new auth state and gets the same error as originally

Reply:
Ah, that makes sense. I had in mind that we wrapped `get_globus_app_with_scopes` with `lru_cache`.